File: lib/mocked/mocke3/src/main/java/com/mobiata/mocke3/RunMockWebServer.java
package com.mobiata.mocke3;

import java.io.File;

import com.squareup.okhttp.mockwebserver.MockWebServer;

public class RunMockWebServer {
	public static void main(String[] args) throws Throwable {
		MockWebServer mockWebServer = new MockWebServer();

		String root = new File("../templates").getCanonicalPath();
		System.out.println("Template path: " + root);
		FileOpener opener = new FileSystemOpener(root);

		ExpediaDispatcher dispatcher = new ExpediaDispatcher(opener);
		mockWebServer.setDispatcher(dispatcher);
		mockWebServer.play(7000);
	}
}

File: lib/mocked/mocke3/src/main/kotlin/com/mobiata/mocke3/RunMockWebServer.kt
package com.mobiata.mocke3

import java.io.File
import java.net.InetAddress

import com.squareup.okhttp.mockwebserver.MockWebServer

public object RunMockWebServer {
	@Throws(Throwable::class)
	@JvmStatic public fun main(args: Array<String>) {
		val server = MockWebServer()

		val root = File("../templates").canonicalPath
		println("Template path: " + root)
		val opener = FileSystemOpener(root)

		val dispatcher = ExpediaDispatcher(opener)
		server.setDispatcher(dispatcher)
		server.start(InetAddress.getByName("0.0.0.0"), 7000)
	}
}
